kubernetes/provider.go

```diff
@@ -100,6 +100,14 @@ func Provider() *schema.Provider {
 				Description:   "Path to the kube config file. Can be set with KUBE_CONFIG_PATH.",
 				ConflictsWith: []string{"config_paths"},
 			},
+			"config_data": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Sensitive:     true,
+				DefaultFunc:   schema.EnvDefaultFunc("KUBE_CONFIG_DATA", nil),
+				Description:   "Raw kube config data for the kubernetes provider",
```

Comment:
initially when running this locally the provider would fail to build due to the descriptions not being the same between `kubernetes/provider.go` and `manifest/provider_config.go`

After making some adjustments as well removing `ConflictsWith`, I was able to produce working applies where i used the `local` provider to test the new field.

A test would be great to include, we'll also want to include this in the kubernetes documentation and finally a changlog generation. Let me know if help is needed to get this PR in a mergable state.